### PR TITLE
Add activity model

### PR DESCRIPTION
**Pull request by ec**

### CODE REVIEW

File: app/models/activity.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Activity < ApplicationRecord
+  CATEGORIES = %w[session user].freeze
+  RESULTS = %w[succeed failed].freeze
+
+  # session -> sign in/out
```
Comment (ec) [inline app/models/activity.rb]: comments should be line before the code they are explaining

File: db/migrate/20181115100105_create_activities.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class CreateActivities < ActiveRecord::Migration[5.2]
+  def change
+    create_table :activities do |t|
+      t.references :user,       null: false
+      t.string     :user_ip,    null: false
+      t.string     :user_agent, null: false
+      t.string     :category,   null: false
```
Comment (mod) [inline db/migrate/20181115100105_create_activities.rb]: rename in topic

File: spec/factories/activity.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :activity do
+  # activity fields cannot be setted in factory
+  # we need specific fields setted correct in every case
```
Comment (mod) [inline spec/factories/activity.rb]: use faker to generate all fields

File: app/models/activity.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+class Activity < ApplicationRecord
+  CATEGORIES = %w[session password otp].freeze
+  RESULTS = %w[succeed failed].freeze
+
+  belongs_to :user
+
+  validates :category, presence: true, inclusion: { in: CATEGORIES }
+  validates :result,   presence: true, inclusion: { in: RESULTS }
+
```
Comment (mod) [inline app/models/activity.rb]: Validate user_ip
Find a way to validate user agent string

include gem https://github.com/fnando/browser
and add methods: .browser .platform / os and more if relevent

File: db/migrate/20181115100105_create_activities.rb (hunk 2 above)
Comment (ec) [reply db/migrate/20181115100105_create_activities.rb]: https://github.com/rubykube/barong2/pull/30/files#diff-1acd2e7e27a227829d5d14a91c863bb6R19

File: spec/factories/activity.rb (hunk 3 above)
Comment (ec) [reply spec/factories/activity.rb]: https://github.com/rubykube/barong2/pull/30/files#diff-3f642a3c942192c5395fdba99e71ba89R3

File: app/models/activity.rb (hunk 4 above)
Comment (ec) [reply app/models/activity.rb]: https://github.com/rubykube/barong2/pull/30/files#diff-c62aeb37fcab74c173b9ae8fde35c5b2R13

**Pull request merged**